src/components/HomePage/SearchArea/SearchArea.js

```diff
@@ -64,8 +147,35 @@ export const AdvancedSearchController = ({
         if (e) e.preventDefault();
         searchOffers(SearchResultsConstants.INITIAL_LIMIT);
 
+        changeURLFilters(location, history, queryParams, { searchValue: queryParams.searchValue ?? searchValue });
+
         if (onSubmit) onSubmit();
-    }, [onSubmit, searchOffers]);
+    }, [changeURLFilters, history, location, onSubmit, queryParams, searchOffers, searchValue]);
+
+    const firstRender = useRef(true);
```

Comment:
This is not the way to do stuff only on first renders. React has a built-in mechanism for that, which is to use a useEffect with an empty deps array `useEffect(fn, [])`

Reply:
I had tested with that option but at the time it was not behaving as I expected: maybe I did something else in a way that produced wrong results and which led me to that solution. Tested the change now and behaves as intended.